lib/iruby/backend.rb

```diff
@@ -50,21 +50,29 @@ def initialize
       @irb = IRB::Irb.new(@workspace)
       @eval_path = @irb.context.irb_path
       IRB.conf[:MAIN_CONTEXT] = @irb.context
-      @completor = IRB::RegexpCompletor.new
     end
 
     def eval_binding
       @workspace.binding
     end
 
     def eval(code, store_history)
-      @irb.context.evaluate(@irb.build_statement(code), 0)
+      if Gem::Version.new(IRB::VERSION) < Gem::Version.new('1.13.0')
+        @irb.context.evaluate(code, 0)
+      else
+        @irb.context.evaluate(@irb.build_statement(code), 0)
+      end
       @irb.context.last_value unless IRuby.silent_assignment && assignment_expression?(code)
     end
 
     def complete(code)
-      # preposing and postposing never used, so they are empty, pass only target as code
-      @completor.completion_candidates('', code, '', bind: @workspace.binding)
+      if defined? IRB::RegexpCompletor # IRB::VERSION >= 1.8.2
+        completor = IRB::RegexpCompletor.new
```

Comment:
It seems that an instance of IRB::RegexpCompletor is created each time. This seems inefficient, what do you think?

Reply:
Indeed, agree! Added [e07b7eb](https://github.com/SciRuby/iruby/pull/358/commits/e07b7eba211ea2ab2d76fa715aa3c9abeb0437e4)